cdap-integration-test/src/main/java/co/cask/cdap/test/remote/RemoteRouteConfigManager.java

```
@@ -0,0 +1,81 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.test.remote;
+
+import co.cask.cdap.client.RouteConfigClient;
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.client.util.RESTClient;
+import co.cask.cdap.common.UnauthenticatedException;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.security.spi.authorization.UnauthorizedException;
+import co.cask.cdap.test.RouteConfigManager;
+import co.cask.common.http.HttpResponse;
+import com.google.common.base.Throwables;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Remote implementation of {@link RouteConfigManager}.
+ */
+public class RemoteRouteConfigManager implements RouteConfigManager {
+  private final NamespaceId namespaceId;
+  private final String appName;
+  private final String serviceName;
+  private final RouteConfigClient routeConfigClient;
+
+  public RemoteRouteConfigManager(NamespaceId namespaceId, String appName, String serviceName,
+                                  ClientConfig clientConfig, RESTClient restClient) {
+    this.namespaceId = namespaceId;
+    this.appName = appName;
+    this.serviceName = serviceName;
+    this.routeConfigClient = new RouteConfigClient(clientConfig, restClient);
+  }
+
+  @Override
+  public Map<String, Integer> getRouteConfig() {
+    try {
+      return routeConfigClient.getRouteConfig(namespaceId, appName, serviceName);
+    } catch (Exception e) {
```

why wrap everything as a runtime exception? what exceptions are thrown here?

---

Trying to follow the style of `RemoteApplicationManager#getHistory` Should I unwrap it and let it throw exceptions directly?